src/main/java/edu/kit/provideq/toolbox/ProblemController.java

```diff
@@ -1,62 +1,112 @@
 package edu.kit.provideq.toolbox;
 
-import edu.kit.provideq.toolbox.meta.MetaSolver;
-import edu.kit.provideq.toolbox.meta.Problem;
-import edu.kit.provideq.toolbox.meta.ProblemSolver;
-import edu.kit.provideq.toolbox.meta.ProblemType;
+import edu.kit.provideq.toolbox.meta.*;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.context.ApplicationContext;
+import org.springframework.context.annotation.Lazy;
 import org.springframework.http.HttpStatus;
+import org.springframework.stereotype.Component;
+import org.springframework.web.bind.annotation.RestController;
 import org.springframework.web.server.ResponseStatusException;
 
+import java.util.List;
+import java.util.Map;
+import java.util.Optional;
 import java.util.Set;
+import java.util.function.Function;
 import java.util.stream.Collectors;
 
 /**
  * Abstract Controller, offers generic post and get methods
- * @param <ProblemFormatType> the type in which problem input is expected to arrive
+ *
+ * @param <ProblemFormatType>  the type in which problem input is expected to arrive
  * @param <SolutionFormatType> the type in which a solution will be formatted
- * @param <SolverType> the type of solver that is to be used to solve a problem
+ * @param <SolverType>         the type of solver that is to be used to solve a problem
  */
+@Component
+@RestController
 public abstract class ProblemController<ProblemFormatType, SolutionFormatType, SolverType extends ProblemSolver<ProblemFormatType, SolutionFormatType>> {
-  public abstract ProblemType getProblemType();
-  public abstract MetaSolver<SolverType> getMetaSolver();
+    @Lazy
+    @Autowired
```

Comment:
Yeah, I know, but I wanted to avoid the mess that is passing injected fields into the super constructor all the time.
Do you still still think we should use the constructor injection for this?